File: code_blocks/integrations/third-party-integrations/firebase-integration_1.kt
import android.app.Application
import com.google.firebase.auth.FirebaseAuth
import com.revenuecat.purchases.Purchases
import com.revenuecat.purchases.logIn

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        Purchases.configure(this, "public_sdk_key")
        FirebaseAuth.getInstance().addAuthStateListener { auth ->
            auth.currentUser?.uid?.let { uid ->
                Purchases.sharedInstance.logIn(uid) { _, _, _ -> }
            }
        }
    }
}

File: code_blocks/integrations/third-party-integrations/firebase-integration_1.ts
import auth from '@react-native-firebase/auth';
import Purchases from 'react-native-purchases';

Purchases.configure({ apiKey: 'public_sdk_key' });

auth().onAuthStateChanged(user => {
  if (user) {
    Purchases.logIn(user.uid);
  }
});
